tests/acceptance/features/bootstrap/WebDav.php

```diff
@@ -1302,28 +1302,28 @@ public function jobStatusValuesShouldMatchRegEx($user, $table) {
 	 * @param string $path
 	 * @param string $type
 	 *
-	 * @return ResponseInterface
+	 * @return void
 	 * @throws \Exception
 	 */
 	public function asFileOrFolderShouldNotExist(
-		$user, $entry, $path, $type = "files"
+		$user, $entry="file", $path= null, $type = "files"
 	) {
 		$user = $this->getActualUsername($user);
-		$path = $this->substituteInLineCodes($path, $user);
-		$response = WebDavHelper::makeDavRequest(
-			$this->getBaseUrl(), $this->getActualUsername($user),
-			$this->getPasswordForUser($user), 'GET', $path,
-			[], null, 2, $type
+		$path = $this->substituteInLineCodes($path);
+		$this->responseXmlObject = $this->listFolder(
+			$user, $path, 0, null, $type
 		);
+		$isCollection = $this->getResponseXmlObject()->xpath("//d:prop/d:resourcetype/d:collection");
 
-		if ($response->getStatusCode() < 401 || $response->getStatusCode() > 404) {
-			throw new \Exception(
-				"$entry '$path' expected to not exist " .
-				"(status code {$response->getStatusCode()}, expected 401 - 404)"
-			);
+		if ($this->isEtagValid()) {
+			if ($entry === "folder") {
+				Assert::assertEquals(\count($isCollection), 0, "Unexpectedly, folder `$path` exists");
+			} else {
+				Assert::assertEquals(\count($isCollection), 1, "Unexpectedly, file `$path` exists");
+			}
```

Comment:
```suggestion
			$isCollection = $this->getResponseXmlObject()->xpath("//d:prop/d:resourcetype/d:collection");
			if (\count($isCollection) === 0) {
				$actualResourceType = "file";
			} else {
				$actualResourceType = "folder";
			}
			Assert::assertFail(
				"$entry '$path' should not exist. But it does exist and is a $actualResourceType";
			);
```

whenever we get inside `isEtagValid()` then we have to fail. Because if the Etag is valid then the file or folder exists.

Code something like this might work to report what type of resource was expected to not exist, and what type of resource did actually exist.

Reply:
fixed